### ISSUE TITLE
Library seems to be failing on some files

### ISSUE THREAD

**Issue by kastashov** [marc1706/fast-image-size#50]
Using the library as part of phpBB 3.2.
One of the users complained that their files were not uploading.
I investigated the issue and found that FastImage Size apparently fails to get image size for their photos.
Interestingly, the core PHP function, getimagesize(), works fine with these files.

Links to files:
http://filedistr.ru/2018/IMG_20180720_103159.jpg
http://filedistr.ru/2018/IMG_20180724_181130.jpg

I fixed the issue on my site by adding the following lines to function TypeJpeg::getSize()

After:

```
		// Look through file for SOF marker
		$size = $this->getSizeInfo();
```
Added:                
```
                if (empty($size)) {
                    $info = getimagesize($filename);
                    if (!empty($info) && is_array($info)) {
                        $size = array(
                            'width' => $info[0],
                            'height' => $info[1],
                        );
                        
                    }
                }
```
This is not necessarily the best way to fix it, this is just so that you have an idea.

**Comment by marc1706** [comment]
What version of phpBB 3.2 and this library are you using? There were some issues with previous versions that could result in the mentioned issue.

**Comment by kastashov** [comment]
I'm using PHPBB 3.2.2 (latest), not too sure abouth the library version, can't find version in code...

**Comment by jdreesen** [comment]
Should be the latest version, according to the `composer.lock`: https://github.com/phpbb/phpbb/blob/release-3.2.2/phpBB/composer.lock#L348-L349

**Comment by kastashov** [comment]
Right. So, the latest version is failing then. Please find links to images that are failing in the ticket description.

**Comment by tinogo** [comment]
Hi, I'm having the same problem with this image: https://www.notendiele.de/uploads/gallery/33.jpg

**Comment by rxu** [comment]
@marc1706 Maybe easy fix (increasing the max header size by 2 or even better by 4) in the meantime?
The bug is quite old, images go ahead by quality/size and phpBB users are facing the issue more often  😃

**Comment by marc1706** [comment]
I've actually invested quite a bit of time investigating this further and testing possible approaches with using streams, However, as remote files do not support seeking (reading at a specific offset) and one of the biggest impacts will also be the latency for remote files I'm inclined to agree.
So IMHO the quickest and yet best solution is really to just increase the maximum header size for JPEG files. I've also arrived at the point where I seriously have to question who came up with this mess of an image format.  🙃

**Comment by FernandoGarcia** [comment]
Hi!
This problem was not solved. I'm facing this problem with PHPBB 3.3 b2 and Fast image size 1.1.5.
Impossible to post this image: https://scooterclub.by/tmp/P90505-140230.jpg. Even changing image size to *786432x1000000*.
The result is :` It was not possible to determine the dimensions of the image. Please verify that the URL you entered is correct.`
Best regards.
